packages/components/src/toolbar/toolbar-item/types.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import type {
+	ReactElement,
+	ReactNode,
+	ElementType,
+	HTMLAttributes,
+	RefAttributes,
+} from 'react';
+
+export interface ToolbarItemProps
```

Comment:
I updated it to use `type` to keep it consistent, but I recommend changing the convention to use `interface` when extending other types.

Intersection types may silently introduce incompatible types that will only show an error at usage. Also, interfaces merge JSDoc better: [TS playground with more details](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgCQCoFkAyBBGMrABGArjAKYDOANHAEpkCGAxjAHIQAmZcAvnAGZQIIOACIojFqIDcAKFkB6BXBgBPMNwDuwGAAs4wAHbkoFMi2ARDFOCAaq4EQhSbEJcEgHMKstRrgAysAANmTGcAC8KBg4eAQk5BQAPGhYACLAAG4AoqEgYTAAfHAAZIiycHBMuiEcEoYA-ABcdJJsnNwAPnAAFD0AlJHFCDz9sjzySiq6DPAM7lbBDgDWhhCaho6bxBQMnmSyTFYU8NW19S1BocYA2qJnwXVhogC6kb2DEcM8corKRiZ+MwtDp9GQAB7kQwcGzaYLBOAUXTrabcIxHcCzIghHQOAieXRzTT2WQAshQIFMbjZcFgYLAJg6OAQqEw6JYXD4IikSgpGIZHJ5ArDCpVGqPerNVrMdpcODdPqfb79OQTP6IiBwTTcJgMTb8YDggwwUnGcmU7gAYWgEhYzMhYTZAHkQDo+Ry4tzEu7MALcmR8sZCjR7uKnoZRCLKg9w1L6DL2HKFQMhohRqrJv8zaZzDBLJs-JQ4AxghRNRwrAByeAzaGhOAAKQCaQgTDgmXJDm18N86m42HeCFFCgAVCPRXAR3A0pQmAQwHmrBOpwABLhA4jBeD4YgHSqThSisBCMAtQgQCChPUZ9XAfharR67eaiAdqAEOU6Gj8aD2hjgUJe38AAhd4BzKIdKlHVd1wYTd4CBUtuBHQ9KmPCBTw8C8r0MG8pj0bgmxbNs4QRBgwA0eY4A4Yg6QZWYyA4XpkTfRxWOAu50LAV4PDIYJ1jGI5rHgQgWg40QuJ4qJELMX4pjJClgTgWsOFCGwmRAYhql4vByVNQElMte1WRsAdILgaC4DXMgNy3AQSzMA8jxPM9sMYXDxkzZTX3JNjfMtTiTx4wg+PWQ5jlOFoAokoK3iiHcyGkIA).

Reply:
Yes, although there is a certain simplicity to sticking with `type` everywhere, I'm open to the "use `interface` unless something requires a `type`" guideline, suggested in the official docs and also pretty much at [Google](https://google.github.io/styleguide/tsguide.html#interfaces-vs-type-aliases).

No need to overhaul anything in bulk, but maybe we can start by preferring `interface` in our own patches and not suggesting changes when a contributor uses `interface`?

Reply:
> No need to overhaul anything in bulk, but maybe we can start by preferring interface in our own patches and not suggesting changes when a contributor uses interface?

Sounds good.

The reason why I suggested `type` is because my personal preference is to opt to consistency when we don't have a guideline.

But I agree with your point! We can definitely agree to got for you approach about `interface` from now on.

Reply:
> fallback to prefer consistency when we don't have a clear guideline on the matter

Me too. Let's try this out low key, and we can add it to the CONTRIBUTING guidelines when we feel a bit more confident about it 😄